internal/util/testutil/equal.go

```diff
@@ -241,9 +241,6 @@ func equalScalars(tb testing.TB, v1, v2 any) bool {
 		if math.IsNaN(s1) {
 			return math.IsNaN(s2)
 		}
-		if s1 == 0 && s2 == 0 {
-			return math.Signbit(s1) == math.Signbit(s2)
```

Comment:
On compat `float64(-123456789)` * 0 return `-0`.
On target `float64(-123456789)` * 0 return `0`.

FerretDB does not support `-0`, and we let target and compat to equate `-0` == `0`